test/jdk/java/awt/TitledBorder/TitledBorderTest.java

```diff
@@ -89,67 +87,66 @@ public static void main(String[] args) throws Exception {
+        boolean checkShadow = false;
+        boolean checkHighlight = false;
+        for (int i = 0; i < img.getWidth(); i++) {
+            int color = img.getRGB(i,y);
+            if (!checkHighlight && !checkShadow) {
+                if (color == shadow.getRGB()) {
+                    checkHighlight = true;
+                    thickness++;
+                } else if (color == highlight.getRGB()) {
+                    throw new RuntimeException("Horizontal Border was clipped or overdrawn.");
+                }
+            } else if (checkHighlight) {
+                if (color == shadow.getRGB()) {
+                    thickness++;
+                } else if (color == highlight.getRGB()) {
+                    verifyThickness(y, thickness, scaling, "Horizontal");
+                    checkHighlight = false;
+                    checkShadow = true;
+                    thickness = 1;
+                } else {
+                    throw new RuntimeException("Horizontal Border has empty space between highlight and shadow.");
+                }
+            } else {
+                if (color == shadow.getRGB()) {
+                    throw new RuntimeException("Border colors reversed.");
+                } else if (color == highlight.getRGB()) {
+                    thickness++;
+                } else {
+                    verifyThickness(y, thickness, scaling, "Horizontal");
+                    checkShadow = false;
+                    thickness = 0;
+                }
+            }
         }
     }
 
-    private static void checkHorizontalBorder(int y, BufferedImage img, double scaling) throws RuntimeException {
-        // checking left border
-        hBorderLoop(0, (int) (Math.floor(scaling)+(5*scaling)), y, img, shadow, scaling);
-        hBorderLoop((int) Math.floor(scaling), (int) (Math.floor(scaling)*2+5*scaling), y, img, highlight, scaling);
-
-        // checking right border
-        hBorderLoop(img.getWidth() - ((int) (Math.floor(scaling)*2+5*scaling)), img.getWidth() - ((int) Math.floor(scaling)),
-                y, img, shadow, scaling);
-        hBorderLoop(img.getWidth() - ((int) (Math.floor(scaling)+5*scaling)), img.getWidth(),
-                y, img, highlight, scaling);
-    }
-
-    private static void verifyThickness(int x, int thickness, double scaling) {
+    private static void verifyThickness(int x, int thickness, double scaling, String orientation) {
         int expected = (int) Math.floor(scaling);
-        if (thickness < expected) throw new RuntimeException("Vertical Border drawn too thin. Thickness: "
-                + thickness + " Scaling: " + scaling + " x: " + x);
-        if (thickness > expected) throw new RuntimeException("Vertical Border drawn too thick. Thickness: "
-                + thickness + " Scaling: " + scaling + " x: " + x);
+        if (thickness != expected) throw new RuntimeException("Unexpected " + orientation + " Border thickness.");
```

Comment:
```suggestion
        if (thickness != expected) {
            throw new RuntimeException("Unexpected " + orientation + " Border thickness.");
        }
```
Please always use braces even for blocks with one statement only.